opensrp-anc/src/main/java/org/smartregister/anc/library/util/Utils.java

```diff
@@ -347,9 +348,8 @@ private static String processValue(String key, Facts facts) {
 
     private static String cleanValueResult(String result) {
         List<String> nonEmptyItems = new ArrayList<>();
-
         for (String item : result.split(",")) {
-            if (item.length() > 0) {
+            if (item.length() > 0 && !item.trim().equals("")) {
```

Comment:
I think you can use the utility method `StringUtils#isNotBlank` for this use case. It will take care of the extra white spaces